pyanaconda/modules/storage/bootloader/bootloader.py

```diff
@@ -456,6 +469,11 @@ def install_bootloader_with_tasks(self, payload_type, kernel_versions):
         :param kernel_versions: a list of kernel versions
         :return: a list of tasks
         """
+        if self._use_bootupd:
```

Comment:
The thing is that setting the bootloader mode to `SKIPPED` is enough to disable the tasks in the `else` branch (there are conditions in their `run` methods that will stop the tasks before any action). So the only effect of the `use_bootupd` flag on this module is running the `bootupctl` tool.

I am wondering if the bootupd code shouldn't stay in the rpm ostree module and the `bootupctl` tool shoudn't be called from there. This functionality is specific to ostree installations, right? Or is there a possibility of expansion to other types of payloads in the future?

Reply:
> This functionality is specific to ostree installations, right?

Today yes.

> Or is there a possibility of expansion to other types of payloads in the future?

In theory it's possible, in practice it'd be a giant change and not really worth it.